src/types.rs

```diff
@@ -142,7 +142,7 @@ impl SumType {
 impl From<SumType> for Type {
     fn from(sum: SumType) -> Type {
         match sum {
-            SumType::Simple { size } => Type::new_simple_predicate(size),
+            SumType::Simple { size } => Type::new_unit_choice(size),
```

Comment:
Reckon you should rename SumType::Simple. Maybe SumType::OfUnits ? (SumType::Unit perhaps but seems more likely to lead to confusion)